### DISCUSSION TITLE
Is linien supposed to generate and show me the error signal?

### DISCUSSION THREAD

**Discussion by doronbehar** [linien-org/linien#328] [Q&A]
Hello :)

First time actually using linien - and I'm not sure if I'm supposed to use it to generate the error signal. My experimental setup is as follows:

![image](https://user-images.githubusercontent.com/10998835/229369615-e82b263f-9132-454f-81ea-473d4b94b93b.png)

I tried first to connect first the photo diode to IN1 and OUT2 to the "current modulation signal" - as shown in the README image:

![setup from readme](https://raw.githubusercontent.com/linien-org/linien/master/docs/setup.png)

And I see no absorption signal in the application. However with the photodiode connected to IN2, I'm able to use the generated sweep signal and view a spectrum with the spectral line I wish to lock my LASER to:

![image](https://user-images.githubusercontent.com/10998835/229369048-3d406c45-0978-443e-879d-6cb8e82e02ee.png)

However I don't understand if linien is supposed to be able to generate the error signal, or should I do it my self somehow. I expected linien would show me both the raw absorption, and the error signal it got by applying the demodulation to the sweep and the PID parameters, but the GUI says that IN2 is supposed to be "error signal"..

**Comment by bleykauf** [comment]
Yes, demodulated signal / error signal is in red. This has nothing to do with PID parameters but rather the modulation/demodulation parameters. Default value is 15 MHz I think so not really suitable for Rb (should be around the natural linewidth).

From your drawing it is not clear how you connect channel IN1 and IN2. IN1 is mandatory (the input signal will be demodulated), IN2 is optional (this is the green line).

**Reply by doronbehar** [reply]
OK so I understand there's no way to view both the original absorption signal and the error signal at the same time.. It would have been nice if this was possible, I'm connected to an oscilloscope in the meantime.

After playing a bit with the modulation and demodulation parameters I managed to view something close to a useful error signal. I was thinking, perhaps something like the following would help?

```diff
:100644 100644 99b5b5b 27a3640 M	linien-gui/linien_gui/ui/modulation_sweep_panel.ui

diff --git a/linien-gui/linien_gui/ui/modulation_sweep_panel.ui b/linien-gui/linien_gui/ui/modulation_sweep_panel.ui
index 99b5b5b..27a3640 100644
--- a/linien-gui/linien_gui/ui/modulation_sweep_panel.ui
+++ b/linien-gui/linien_gui/ui/modulation_sweep_panel.ui
@@ -28,7 +28,7 @@
         <item>
          <widget class="CustomDoubleSpinBoxNoSign" name="modulation_frequency">
           <property name="decimals">
-           <number>5</number>
+           <number>0</number>
           </property>
          </widget>
         </item>
@@ -44,7 +44,7 @@
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
-         <string>Set modulation frequency to 0 to display the DC signal.</string>
+         <string>Set to 0 to display in the "error signal" the system's response to the unmodulated sweep.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
```

I read that string when I tried to get something to work before I created this help thread, and I didn't understand what "DC signal" means..
